sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java

```diff
@@ -105,7 +107,30 @@ public final class ReflectionSerializable {
      * @return Whether {@code bodyContentClass} can be used as {@code JsonSerializable}.
      */
     public static boolean supportsJsonSerializable(Class<?> bodyContentClass) {
-        return JsonSerializable.class.isAssignableFrom(bodyContentClass);
+        if (!JsonSerializable.class.isAssignableFrom(bodyContentClass)) {
+            return false;
+        }
+
+        boolean hasFromJson = false;
+        boolean hasToJson = false;
+        for (Method method : bodyContentClass.getDeclaredMethods()) {
```

Comment:
This is actually the opposite. This change makes it so the subclasses of `ProxyResource` and `Resource` won't use `JsonSerializable` until they themselves implement `toJson` and `fromJson` to allow for a deferred migration to stream-style serialization.

Since `ProxyResouce` and `Resource` are still maintaining their Jackson annotations and the sub types haven't been migrated yet they'll continue to use Jackson.

For example:

```java
public class Resource implements JsonSerializable<Resource> {
    // impl stuff

    @Override
    public JsonWriter toJson(JsonWriter writer) throws IOException {
        // impl
    }
   
    public static Resource fromJson(JsonReader reader) throws IOException {
        // impl
    }
}
```

Since `Resource` declares `toJson` and `fromJson` it will use JsonSerializable for serialization.

```java
public class SubResource extends Reource {
    // impl stuff
}
```

Since `SubResource` doesn't declare `toJson` or `fromJson` it will use Jackson for serialization.

Reply:
My bad, thought `getDeclaredMethods` includes those inherited. Then this is good.